src/OrchardCore/OrchardCore.DisplayManagement/Handlers/DisplayDriver.cs

```diff
@@ -92,24 +87,19 @@ public virtual IDisplayResult Display(TModel model)
 
         public virtual Task<IDisplayResult> EditAsync(TModel model, TEditorContext context)
         {
-            return EditAsync(model, context.Updater);
+            return Task.FromResult(Edit(model, context));
         }
 
-        public virtual Task<IDisplayResult> EditAsync(TModel model, IUpdateModel updater)
```

Comment:
Removing this is now an impactful breaking change, as we can also see from all the drivers needing adaptation. And I don't think the end result is better: apparently, implementations frequently need just `IUpdateModel` and having to get it via `TEditorContext` is complicating things. 

Overall, the drawbacks now outweigh the benefits I think. You get the sync overloads for cases where you need them, but that's paying a runtime price due to the additional virtual method call to get a code editing time benefit. I don't see this being worth it.

Having to catch up with breaking changes yet again in the solutions @sarahelsaig is updating to the preview isn't appealing either.

Reply:
The signature will change, but the implementation does not. I actually think utilizing the `context` is better because we get everything we want without having to worry about many arguments or many methods.

IMO, having `Edit(TModel model, TEditorContext context)` and `Display(TModel model, TDisplayContext context)` are more important than `EditAsync(TModel model, IUpdateModel updater)`.

The end result, these two methods will help us cleaning up drivers.

Reply:
When would you use the sync methods, though? I count 8 instances in this PR. This is not insignificant, but at the same time, we have more than 330 `DisplayDriver` implementations.

And then again, what's really the goal? Having sync methods is nice to have, but at the same time, do we really want to add this? It adds a runtime overhead and breaking change for a bit nicer code for the minority of cases. It's also not without sublte runtime risks that are much harder to detect than a build error.

Having breaking changes already is no reason to add more ;). This is already the costliest OC upgrade, by at least an order of magnitude if not two, that we all have to pay for every Orchard Core app there. We shouldn't add more on top of it without really good reasons. I don't see this as a good enough reason.

Reply:
> Having sync methods is nice to have

We are not adding Async methods. Instead we are adding `Display()` and `Edit()` methods. I agree there isn't huge benefit, but it cleans things up specially if you have project with lots of drivers. Personally, I still think this is a good change to take. But the way, if we don't we have to revert PR #16501 .

> Having breaking changes already is no reason to add more ;

Correct. But since we have a breaking change this is the best time to make changes.

Reply:
I've written having **sync** methods is nice to have. Async ones are necessary, but sync ones aren't. Why would we need to revert https://github.com/OrchardCMS/OrchardCore/pull/16501?

Reply:
Because that new method will never be called. It's something this PR will also fix.

Reply:
Then that's only what needs to be fixed. I don't see how adding sync methods would help with that.

Reply:
To fix that, we have to remove `public virtual Task<IDisplayResult> EditAsync(TModel model, IUpdateModel updater)` which is what this PR is doing. And doing the same thing for `DisplayAsync` which is cleaner.

Reply:
@Piedone I updated the PR to show the invocation path from before and then after this PR. I think it is more simplified now and also the `Display` path is one less call since we remove the weird `IUpdateModel` on the display path.